cypress/e2e/cloud/pinned.test.ts

```diff
@@ -0,0 +1,330 @@
+import {Organization} from '../../../src/types'
+import {createFirstTask} from '../shared/tasks.test'
+
+describe('Pinned Items', () => {
+  let orgID: string
+  beforeEach(() => {
+    cy.flush()
+
+    cy.signin().then(() =>
+      cy.fixture('routes').then(({orgs}) => {
+        cy.get('@org').then(({id}: any) => {
+          orgID = id
+          cy.setFeatureFlags({
+            pinnedItems: true,
+            docSearchWidget: true,
+          }).then(() => {
+            cy.wait(1000)
+            cy.visit(`${orgs}/${id}`)
+            cy.getByTestID('tree-nav')
+          })
+        })
+      })
+    )
+  })
+
+  it('renders a pinned items modal in the homepage with initial empty state', () => {
+    cy.getByTestID('pinneditems--emptystate').should(
+      'contain.text',
+      'Pin a task, dashboard, or notebook here'
+    )
+  })
+
+  describe('Pin dashboard tests', () => {
+    beforeEach(() => {
+      cy.createDashboard(orgID).then(() => {
+        cy.setFeatureFlags({
+          pinnedItems: true,
+          docSearchWidget: true,
+        }).then(() => {
+          cy.fixture('routes').then(({orgs}) => {
+            cy.visit(`${orgs}/${orgID}/dashboards-list`)
+            cy.getByTestID('tree-nav')
+          })
+        })
+      })
+    })
+    it('pins a dashboard to the homepage for easy access as a pinned item', () => {
+      cy.getByTestID('dashboard-card')
+        .first()
+        .trigger('mouseover')
+        .within(() => {
+          cy.getByTestID('context-pin-menu').click()
+          cy.getByTestID('context-pin-dashboard').click()
+        })
+      cy.visit('/')
+      cy.getByTestID('pinneditems--container').within(() => {
+        cy.getByTestID('pinneditems--card').should('exist')
+      })
+    })
+
+    it('reflects an edit to the dashboard name on the dashboard card', () => {
+      cy.getByTestID('dashboard-card')
+        .first()
+        .trigger('mouseover')
+        .within(() => {
+          cy.getByTestID('context-pin-menu').click()
+          cy.getByTestID('context-pin-dashboard').click()
+        })
+      cy.getByTestID('dashboard-card').within(() => {
+        cy.getByTestID('dashboard-card--name')
+          .first()
+          .trigger('mouseover')
+
+        cy.getByTestID('dashboard-card--name-button')
+          .first()
+          .click()
+
+        cy.get('.cf-input-field')
+          .type('Bucks In Six')
+          .type('{enter}')
+      })
+
+      cy.visit('/')
```

Comment:
I would add a `cy.getByTestID('nav-tree')` here after the `cy.visit('/')` so the test waits for the page to load before continuing. Should reduce flakes.